snowfakery/data_generator_runtime.py

```diff
@@ -207,12 +207,18 @@ def register_intertable_reference(
             Dependency(table_name_from, table_name_to, fieldname)
         )
 
-    def reset_slots(self):
+    def reset_slots(self, nicknamed_objects: Dict = None):
         "At the beginning of every execution, reset the forward reference slots"
         self.named_slots = {
             name: NicknameSlot(table, self.id_manager)
             for name, table in self.nicknames_and_tables.items()
         }
+        if nicknamed_objects:
+            nicknames_with_slots = set(nicknamed_objects.keys()).intersection(
+                self.named_slots.keys()
+            )
```

Comment:
I just experimented and found that this can also be written as `{*nicknamed_objects}.intersection(self.named_slots)`

The way you have it is probably more obvious though.